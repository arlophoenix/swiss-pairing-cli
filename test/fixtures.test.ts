import { execSync } from 'child_process';
import fs from 'fs';
import path from 'path';

const runCLI = (args: string) =>
  execSync(`node dist/index.js ${args}`, {
    encoding: 'utf-8',
  });

describe('Fixtures', () => {
  const fixturesDir = path.join(__dirname, 'fixtures');

  fs.readdirSync(fixturesDir).forEach((fixture) => {
    test(`CLI Output - ${fixture}`, () => {
      const input = fs.readFileSync(path.join(fixturesDir, fixture), 'utf-8');
      const output = runCLI(input);
      expect(output).toMatchSnapshot();
    });
  });
});
