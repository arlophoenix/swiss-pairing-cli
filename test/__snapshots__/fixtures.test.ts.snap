// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixtures CLI Output - Round6-1Unmodified.txt 1`] = `
"Matches generated successfully: {"Round 6":[["Morgath","Pauldog"],["Nzdaemon","GreyGhost"],["Saravock","lilme"],["Arlo","Colain"],["armydan","LCYig"],["iso.evil","Clin"]]}
"
`;

exports[`Fixtures CLI Output - Round6-2GGWin.txt 1`] = `
"Matches generated successfully: {"Round 6":[["Morgath","Pauldog"],["Nzdaemon","GreyGhost"],["Saravock","lilme"],["Arlo","Colain"],["armydan","LCYig"],["iso.evil","Clin"]]}
"
`;

exports[`Fixtures CLI Output - Round6-3ADWin.txt 1`] = `
"Matches generated successfully: {"Round 6":[["Morgath","armydan"],["Nzdaemon","Pauldog"],["Saravock","lilme"],["Arlo","LCYig"],["GreyGhost","Colain"],["iso.evil","Clin"]]}
"
`;

exports[`Fixtures CLI Output - Round6-4Draw.txt 1`] = `
"Matches generated successfully: {"Round 6":[["Morgath","Pauldog"],["Nzdaemon","GreyGhost"],["Saravock","lilme"],["Arlo","Colain"],["armydan","LCYig"],["iso.evil","Clin"]]}
"
`;
